rotkehlchen/chain/ethereum/modules/eth2.py

```diff
@@ -386,24 +324,75 @@ def get_validator_daily_stats(
+            field = 'validator_index'
+            valid_index = validator_index
+            valid_pubkey = public_key
+            if dbeth2.validator_exists(field=field, arg=valid_index):
+                raise InputError(f'Validator {valid_index} already exists in the DB')
+        else:  # we are missing one of the 2
+            if validator_index is None:
+                field = 'public_key'
+                arg = public_key
+            else:  # we should have valid index
+                field = 'validator_index'
+                arg = validator_index  # type: ignore
+
+            if dbeth2.validator_exists(field=field, arg=arg):  # type: ignore
+                raise InputError(f'Validator {arg} already exists in the DB')
+
+            # at this point we gotta query for one of the two
+            result = self.beaconchain._query(
+                module='validator',
+                endpoint=None,
+                encoded_args=arg,  # type: ignore
+            )
+            if not isinstance(result, dict):
+                raise RemoteError(
+                    f'Validator data for {arg} could not be found. Likely invalid validator.')
+
+            try:
+                valid_index = result['validatorindex']
+                valid_pubkey = Eth2PubKey(result['pubkey'])
+            except KeyError as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key entry for {msg}.'
+
+                raise RemoteError(f'Failed to query beaconcha.in for validator data due to: {msg}') from e  # noqa: E501
+
+        # by now we have a valid index and pubkey. Add to DB
+        dbeth2.add_validators([Eth2Validator(index=valid_index, public_key=valid_pubkey)])
+
     # -- Methods following the EthereumModule interface -- #
     def on_startup(self) -> None:
         pass
 
-    def on_account_addition(self, address: ChecksumEthAddress) -> Optional[List['AssetBalance']]:
+    def on_account_addition(self, address: ChecksumEthAddress) -> Optional[List[AssetBalance]]:  # pylint: disable=useless-return  # noqa: E501
+        """Just query balances and add detected validators to DB. Return nothing"""
         try:
-            result = self.get_balances([address])
+            self.get_balances(
+                addresses=[address],
+                fetch_validators_for_eth1=True,
+            )
         except RemoteError as e:
             self.msg_aggregator.add_error(
                 f'Did not manage to query beaconcha.in api for address {address} due to {str(e)}.'
                 f' If you have Eth2 staked balances the final balance results may not be accurate',
             )
-            return None
-        balance = result.get(address)
-        if balance is None:
-            return None
 
-        return [AssetBalance(asset=A_ETH2, balance=balance)]
+        return None
 
     def on_account_removal(self, address: ChecksumEthAddress) -> None:
         pass
```

Comment:
Shouldn't information for validators be removed here?

Reply:
on the fence about it. I am more leaning towards no. Since you can remove them anyway via the validators remove/add/get endpoints.